internal/types/types.go

```diff
@@ -23,6 +27,24 @@ func (u UniquePath) String() string {
 	return string(u)
 }
 
+// FromSlashPath returns a UniquePath according to the input slash 'path'.
+// 'base' should be an OS-specific base path which will be joined with 'path'
+// if 'path' is not absolute.
+func FromSlashPath(base, path string) (UniquePath, error) {
```

Comment:
Not sure if this method is required generally to be put in this pkg. The logic seems to be very custom. My recommendation, try keeping it closer to where it was needed, if needed again, then move it common place.